pkg/endpoints/pipeline.go

```diff
@@ -166,6 +212,112 @@ func (r Resource) getPipelineRun(request *restful.Request, response *restful.Res
 	response.WriteEntity(pipelinerun)
 }
 
+func (r Resource) createPipelineRun(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter("namespace")
+
+	pipelineRunData := ManualPipelineRun{}
+	if err := request.ReadEntity(&pipelineRunData); err != nil {
+		logging.Log.Error("error parsing request body on call to createPipelineRun %s", err)
+		utils.RespondError(response, err, http.StatusBadRequest)
+		return
+	}
+
+	createResponse := r.CreatePipelineRunImpl(pipelineRunData, namespace)
+	if createResponse.CODE == 204 {
+		response.WriteHeader(http.StatusNoContent)
+	} else { // anything other than 204 is an error - RespondError
+		utils.RespondError(response, createResponse.ERROR, createResponse.CODE)
+	}
+}
+
+// CreatePipelineRunImpl ...
+/* Create a new manual pipeline run and resources in a given namespace */
+func (r Resource) CreatePipelineRunImpl(pipelineRunData ManualPipelineRun, namespace string) *AppError {
+
+	pipelineName := pipelineRunData.PIPELINENAME
+
+	saName := ""
+	providedSaName := os.Getenv("PIPELINE_RUN_SERVICE_ACCOUNT")
+	if providedSaName != "" {
+		saName = providedSaName
+	} else {
+		saName = "default"
+	}
```

Comment:
Can we move this `provideSaName` logic into a reusable function? It seems like "general purpose" logic. I think the control flow of using the `default` service account name if the `PIPELINE_RUN_SERVICE_ACCOUNT` does not exist should be documented in a general location, such as the utils.